pingcap/tidb
# planner: fix a bug that point get plan returns wrong column name

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/23291

as https://github.com/pingcap/tidb/issues/23291#issuecomment-799042832 says, `the results of TiDB are problematic`
also, we explicitly select the column name and expect to get the result with the same column name.

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)
```sql
create table t1(`COL1` varchar(255), PRIMARY KEY (`COL1`));
insert into t1 values("a");
SELECT col1 FROM t1 where col1="a";
+------+
| col1 |    -- it should be lower case
+------+
| a    |
+------+
1 row in set
```

### Release note 

- fix a bug that point get plan returns wrong column name
